components/automate-ui/src/app/modules/desktop/insight/insight.component.ts

```diff
@@ -50,4 +76,31 @@ export class InsightComponent {
   public updateFilters(event: FilterUpdate) {
     console.log(event);
   }
+
+  toggleDownloadDropdown() {
+    this.downloadOptsVisible = !this.downloadOptsVisible;
+  }
+
+  hideDownloadDropdown() {
+    this.downloadOptsVisible = false;
+  }
+
+  onDownloadOptPressed(format) {
+    this.downloadOptsVisible = false;
+
+    const filename = `${moment().utc().format(DateTime.REPORT_DATE_TIME)}.${format}`;
+
+    const onComplete = () => this.downloadInProgress = false;
+    const onError = _e => this.downloadFailed = true;
+    const types = {'json': 'application/json', 'csv': 'text/csv'};
+    const onNext = data => {
+      const type = types[format];
+      const blob = new Blob([data], {type});
+      saveAs(blob, filename);
+    };
+
+    this.requests.downloadNodes(format, this.nodeFilter).pipe(
+      finalize(onComplete))
+      .subscribe(onNext, onError);
```

Comment:
just in case - is this something you need to unsubscribe from?

Reply:
(1) Second to @SEAjamieD's question--I believe you do need the standard `takeUntil(this.isDestroyed)` boilerplate for this subscription, plus the one on L50.

(2) Is there a reason to use `finalize` instead of the 3-arg `subscribe`?
```
.subscribe(onNext, onError, onComplete);
```